spec/activerecord_hoarder/batch_archiver_spec.rb

```diff
@@ -1,10 +1,73 @@
 require 'spec_helper'
 
 RSpec.describe ::ActiverecordHoarder::BatchArchiver do
-  let(:model_class) { double("Record", table_name: "records") }
+  subject { ::ActiverecordHoarder::BatchArchiver.new(ExampleHoarder, storage) }
+  let(:batch_collector) { double("@batch_collector") }
+  let(:batch_instance) { double("batch_instance", delete_records!: nil) }
+  let(:storage) { double("storage") }
+
+  before do
+    allow(::ActiverecordHoarder::BatchCollector).to receive(:new).and_return(batch_collector)
```

Comment:
Try to stay away from mocking the methods of the class you are testing. Better option is to inject a dependency via `new`

Reply:
I don't quite understand. Of course it seems nonsensical to mock the methods you are testing. It seems like you'd be testing the mock rather than the method. I don't believe this to be the case here, but I do want to make sure to avoid it. I have been wondering about testing patterns. I have the impression that writing unit tests simply tests the method being described, not the encapsulated logic in the methods called